packages/beacon-transport-walletconnect/src/communication-client/WalletConnectCommunicationClient.ts

```diff
@@ -422,10 +395,16 @@ export class WalletConnectCommunicationClient extends CommunicationClient {
       // therefore we must immediately open a session
       // to get data required in the pairing response
       try {
-        const session = await this.openSession(topic)
+        let session = await this.openSession(topic)
+
+        if (!session.sessionProperties) {
+          const fun = this.eventHandlers.get(ClientEvents.WC_ACK_NOTIFICATION)
+          fun && fun()
+          this.requestAccountNamespacePromise = new ExposedPromise()
+          await this.requestAccountNamespacePromise?.promise
+          session = this.getSession()
+        }
```

Comment:
This code looks very similar to the one in the `requestPermissions` method. Shouldn't only one of those 2 be needed?